app/src/main/java/com/example/ar_reshare/ProductPageActivity.java

```diff
@@ -147,19 +163,95 @@ public void run() {
+                    public boolean onMenuItemClick(MenuItem item) {
+                        switch (item.getTitle().toString()){
+                            case "Edit":
+                                item.setEnabled(false);
+                                int picCount = product.getPictures().size();
+                                imageUriList = new String[picCount];
+                                for (int i = 0; i < picCount; i++) {
+                                    imageUriList[i] = getImageUri(getApplicationContext(),product.getPictures().get(i)).toString();
+                                }
+                                Intent intent = new Intent(ProductPageActivity.this, ModifyProduct.class);
+                                intent.putExtra("productID", product.getId());
+                                intent.putExtra("productName", product.getName());
+                                intent.putExtra("productDescription", product.getDescription());
+                                intent.putExtra("categoryID",product.getCategoryID());
+                                intent.putExtra("condition",product.getCondition());
+                                intent.putExtra("postcode", postcode);
+                                intent.putExtra("images",imageUriList);
+                                startActivity(intent);
+                                return true;
+                            case "Delete":
+                                item.setEnabled(false);
+                                confirmToDelete();
+                                return true;
+                            default:
+                                return false;
+                        }
+                    }
+                });
+                popupMenu.show();
+            }
+        });
+    }
+
+    private void confirmToDelete(){
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Do you want to delete this product?")
+                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        try {
+                            BackendController.closeListing(product.getId(), new BackendController.BackendCallback() {
+                                @Override
+                                public void onBackendResult(boolean success, String message) {
+                                    if(success){
+                                        Toast.makeText(getApplicationContext(),
+                                                "Product deleted successfully!",
+                                                Toast.LENGTH_LONG).show();
+                                        // go back to the main page when finished
+                                        startActivity(new Intent(ProductPageActivity.this, ARActivity.class));
```

Comment:
Are you sure going back to main activity after deleting the item is the best choice? Should we maybe go to profile page? What does everyone think?

Reply:
I feel like we should go to the profile page as well!

Reply:
yeah that makes more sense, thanks guys! I'll change it once profile is ready